JAICore/jaicore-ml/src/main/java/jaicore/ml/core/evaluation/measure/EventEmittingMeasure.java

```diff
@@ -9,39 +9,55 @@
 import jaicore.ml.evaluation.MeasureListComputationEvent;
 import jaicore.ml.evaluation.MeasureSingleComputationEvent;
 
-public class EventEmittingMeasure<INPUT, OUTPUT> implements IMeasure<INPUT, OUTPUT> {

```

Comment:
I understand the idea here, but wouldn't it be more nautral that this is an interface? The problem I see here with using the decorator pattern is that all the interfaces implemented by the base measure get "lost".

My suggestion would be to have this an interface and use an abstract class to provide default implementations.